mp/src/game/client/momentum/ui/HUD/hud_mapfinished.cpp

```diff
@@ -425,25 +443,85 @@ void CHudMapFinishedDialog::SetCurrentPage(int pageNum)
     //MOM_TODO: Set every label's Y pos higher if there's no ZoneEnterTime visible
 
     m_pZoneJumps->SetText(CConstructLocalizedString(m_iCurrentPage == 0 ? m_pwJumpsOverall : m_pwJumpsZone,
-                                                    m_pRunStats ? m_pRunStats->GetZoneJumps(m_iCurrentPage) : 0));
+                                                    m_pRunStats->GetZoneJumps(m_iCurrentPage)));
 
     m_pZoneStrafes->SetText(CConstructLocalizedString(m_iCurrentPage == 0 ? m_pwStrafesOverall : m_pwStrafesZone,
-                                                      m_pRunStats ? m_pRunStats->GetZoneStrafes(m_iCurrentPage) : 0));
+                                                      m_pRunStats->GetZoneStrafes(m_iCurrentPage)));
 
     m_pZoneSync1->SetText(CConstructLocalizedString(m_iCurrentPage == 0 ? m_pwSync1Overall : m_pwSync1Zone,
-                                                    m_pRunStats ? m_pRunStats->GetZoneStrafeSyncAvg(m_iCurrentPage) : 0.0f));
+                                                    m_pRunStats->GetZoneStrafeSyncAvg(m_iCurrentPage)));
 
     m_pZoneSync2->SetText(CConstructLocalizedString(m_iCurrentPage == 0 ? m_pwSync2Overall : m_pwSync2Zone,
-                                                    m_pRunStats ? m_pRunStats->GetZoneStrafeSync2Avg(m_iCurrentPage) : 0.0f));
+                                                    m_pRunStats->GetZoneStrafeSync2Avg(m_iCurrentPage)));
 
     int velType = m_cvarVelType.GetInt();
-    m_pZoneVelEnter->SetText(CConstructLocalizedString(m_pwVelZoneEnter, m_pRunStats ? m_pRunStats->GetZoneEnterSpeed(m_iCurrentPage, velType) : 0.0f));
+    m_pZoneVelEnter->SetText(CConstructLocalizedString(m_pwVelZoneEnter, m_pRunStats->GetZoneEnterSpeed(m_iCurrentPage, velType)));
+
+    m_pZoneVelExit->SetText(CConstructLocalizedString(m_pwVelZoneExit, m_pRunStats->GetZoneExitSpeed(m_iCurrentPage, velType)));
+
+    m_pZoneVelAvg->SetText(CConstructLocalizedString(m_pwVelAvg, m_pRunStats->GetZoneVelocityAvg(m_iCurrentPage, velType)));
+
+    m_pZoneVelMax->SetText(CConstructLocalizedString(m_pwVelMax, m_pRunStats->GetZoneVelocityMax(m_iCurrentPage, velType)));
+}
+
+void CHudMapFinishedDialog::LoadPlayerBestTime()
+{
+    const auto pPlayer = C_MomentumPlayer::GetLocalMomPlayer();
+    if (!pPlayer)
+        return;
+
+    const auto pRunData = pPlayer->GetCurrentUIEntData();
+
+    CMomReplayBase *pBase = MomUtil::GetBestTime(g_pGameRules->MapName(), pRunData->m_flTickRate,
+                                                 pRunData->m_iCurrentTrack, pRunData->m_iRunFlags);
+
+    if (!pBase)
+        return;
+    
+    m_pPbRunStats.FullyCopyFrom(*pBase->GetRunStats());
```

Comment:
Ah, lastly we're gonna need a way to "clear" `m_PbRunStats` if the player's PB fails to load but `m_PbRunStats` is still valid (has a PB on one map but doesn't on another).

There should be a `m_pPbRunStats.Init()` that you can call if the PB fails to load.